### PR TITLE
[ckpt loader] Allow loading the Inpaint and Img2Img pipelines, while loading a ckpt model

### PR DESCRIPTION
**Pull request by cmdr2**
Hi, this PR allows us to load the Inpainting and Img2Img pipelines, while loading a ckpt/safetensors file. I've tested that it works, and performs inpainting and img2img correctly. Please let me know if you'd like any changes!

Side note: I'd be happy to reorganize the repetitive pipeline arguments in the if-else chain, but wanted to keep this PR focused on the specific change, instead of a mass refactor. Could use a common dict for the args, and flatten (using `**`) that dict while creating the pipeline. But that can be a separate PR, if you're interested.

As a background, I'm looking to switch the backend of our project to diffusers (for [Easy Diffusion](https://github.com/cmdr2/stable-diffusion-ui), previously cmdr2 UI). Inpainting and Img2Img via ckpt is a very frequent operation for our users.

Thanks for this in-memory ckpt converter! It's really useful, and (the lack of it) was the main reason why we didn't switch to diffusers earlier. :)

Cheers
~cmdr2

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py
Comment (cmdr2) [reply src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py:1198]: Sure, I'm okay with that change. My only concern is if a different model in the future (unrelated to inpainting) also uses 8 channels. I'll defer to your knowledge here, since my understanding is limited. Maybe 8 channels always signify inpainting?

The links to the yaml config files are at:
* 1.5-inpainting: https://github.com/runwayml/stable-diffusion/blob/main/configs/stable-diffusion/v1-inpainting-inference.yaml#L3
* 2.0-inpainting: https://github.com/Stability-AI/stablediffusion/blob/main/configs/stable-diffusion/v2-inpainting-inference.yaml#L3

I'm okay with whatever you suggest, happy to make the change. Thanks!

Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py:1198]:
```suggestion
                if hasattr(original_config, "model") and hasattr(original_config.model, "target") and "LatentInpaintDiffusion" in original_config.model.target:
```

Can we make it a bit more "defensive" here in case a weird config is passed ? :-) (maybe below as well)

**Review by patrickvonplaten — APPROVED** [review]
Looks good to me, just left a small comment. 

@williamberman @sayakpaul @yiyixuxu could you check as well maybe?

File: src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py
Comment (sayakpaul) [reply src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py:1198]: 8 channels actually kind of imply that the underlying pipeline has another spatial conditioning (can be a real image, can be a mask, etc.).

**Comment by cmdr2** [comment]
Thanks @patrickvonplaten and @sayakpaul for the review! I've made the change that was requested.

We've started using `diffusers` successfully in the beta build of Easy Diffusion, and the users are quite happy with it. https://github.com/cmdr2/stable-diffusion-ui/wiki/The-beta-channel

I'm currently using a copy of the submitted script in our repo, until it's merged into diffusers.

**Comment by cmdr2** [comment]
Hi, just checking if anything's still required from my end on this PR. The requested changes have been made.

The MPS test is unrelated to this PR, and the code quality test is failing for lines that aren't part of this PR. I'd be happy to lint the code and submit a separate PR, but those lines aren't part of this PR.

Thanks!

**Comment by sayakpaul** [comment]
@cmdr2 the failing tests are unrelated. @patrickvonplaten I think we are good to merge this one?

**Pull request merged**
